public/js/controllers/charts_controller.js

```diff
@@ -629,6 +708,42 @@ export default class extends Controller {
     this._zoomCallback(start, end)
   }
 
+  toggleAllControls (isAllDisabled) {
+    this.toggleBinControls(isAllDisabled)
+    this.toggleAxisControls(isAllDisabled)
+    this.toggleScaleControls(isAllDisabled)
+    this.toggleZoomLabelControls(isAllDisabled)
+  }
+
+  toggleVisibilityControls (isVEnabled) {
+    var vControl = this.vSelectorTarget.classList
+    isVEnabled ? vControl.remove('d-hide') : vControl.add('d-hide')
+  }
+
+  toggleScaleControls (isScaleDisabled) {
+    var scaleControl = this.scaleSelectorTarget.classList
+    isScaleDisabled ? scaleControl.add('d-hide') : scaleControl.remove('d-hide')
+  }
+
+  toggleZoomLabelControls (isZoomDisabled) {
+    var zoomControl = this.zoomLabelTarget.classList
+    var limitControl = this.limitLabelTarget.classList
+    isZoomDisabled ? zoomControl.add('d-hide') : zoomControl.remove('d-hide')
+    isZoomDisabled ? limitControl.remove('d-hide') : limitControl.add('d-hide')
+  }
+
+  toggleBinControls (isBinControls) {
+    var binControl = this.binSelectorTarget.classList
+    isBinControls ? binControl.add('d-hide') : binControl.remove('d-hide')
+  }
+
+  toggleAxisControls (isAxesControls) {
+    var dualAxes = this.dualAxesSelectorTarget.classList
+    var customAxis = this.customAxisSelectorTarget.classList
+    isAxesControls ? dualAxes.add('d-hide') : dualAxes.remove('d-hide')
+    isAxesControls ? customAxis.remove('d-hide') : customAxis.add('d-hide')
+  }
+
```

Comment:
Can we maybe reduce all of these functions to just one function that takes a list of elements to show and hides all the rest?